compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -5336,7 +5336,7 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
 
             if (type != SyntaxKind.TEMPLATE_STRING && type != SyntaxKind.XML_TEXT_CONTENT) {
                 try {
-                    text = IdentifierUtils.unescapeJava(IdentifierUtils.unescapeUnicodeCodepoints(text));
+                    text = IdentifierUtils.unescapeJava(IdentifierUtils.unescapeUnicodeCodepoints(text, true));
```

Comment:
Shall we create a new method in `IdentifierUtils` which perform the operation done by both of these function calls. the reason for this is only public use case for `unescapeUnicodeCodepoints` with 2 param is this one. So let's make `unescapeUnicodeCodepoints` private and let's give a meaningful name to the new method.

Reply:
Gave the name as `unescapeBallerina` similar to `unescapeJava` method in the same class.